tools/form-generation-tool.adoc

```diff
@@ -0,0 +1,188 @@
+
+
+Browsing linked examples, you may notice the projects use Maven. So when Maven builds an https://incubator.apache.org/projects/kie.html[Apache Kie] project, Kie generates JSON schemas to represent the data models for both Workflows and User Tasks.
+Then *{FORM_GENERATION_TOOL}* locates those JSON Schemas in the project and taking advantage of the Uniforms APIs, processes them and generates static forms as a resource in the project `src/main/resources/forms` folder.
+
+*Generation Prerequisities*
+
+- Make sure your https://incubator.apache.org/projects/kie.html[Apache Kie] project has been compiled, otherwise the CLI won't be able to find all the form schemas.
+
+As we installed the *{FORM_GENERATION_TOOL}* globally, we can run the command from anywhere.
+[source,bash]
+----
+form-generation-tool
+----
+
+This will start a wizard to help you generate the forms:
+
+First, set the path to your https://incubator.apache.org/projects/kie.html[Apache Kie] Project. We will use https://github.com/apache/incubator-kie-kogito-examples/tree/stable/kogito-quarkus-examples/process-quarkus-example[process-quarkus-example].
+
+[source,bash]
+----
+Kogito Form Generation CLI
+===========================
+ 
+This tool will help you generate forms for the Processes and User Tasks in your Kogito Projects.
+The tool will search for the User Tasks JSON schemas generated in your project, so make sure the project is build.
+The generated forms will be stored as resources in your project (in src/main/resources/forms folder).
+ 
+? Type your Kogito Project path:---> /path/to/my/projects/process-quarkus-example <---
+----
+
+Select one of the available Form types (Patternfly or Bootstrap 4). We will use Bootstrap 4. Use arrow keys to change the selection.
+
+[source,bash]
+----
+? Select the Form type:
+  patternfly
+❯ bootstrap
+----
+
+Confirm Selection, type *Y*, to start the Form Generation process.
+
+[source,bash]
+----
+Current selection:
+Project path: /path/to/my/projects/process-quarkus-example
+Form type: bootstrap
+ 
+? Do you want to continue? (Y/n) Y
+----
+
+Once the generation is done, let's browse `/path/to/my/projects/process-quarkus-example/src/main/resources/forms` folder.
+
+[source,bash]
+----
+ls /path/to/my/projects/process-quarkus-example/src/main/resources/forms/
+ 
+demo_orderItems.config
+demo_orderItems.html
+demo_orderItems_Verify_order.config
+demo_orderItems_Verify_order.html
+demo_orders.config
+demo_orders.html  
+persons.config
+persons.html
+persons_ChildrenHandling.config
+persons_ChildrenHandling.html
+----
+
+One question that may come to your mind, is why there are so many forms. We have only three Workflows in `process-quarkus-example` project. 
+Well spotted. It is because not only Workflows can have their forms, but also User Tasks, as special nodes that require input from users, can have their own forms.
```

Comment:
User Tasks also have forms generated.
Even though there only three Workflows in the `process-quarkus-example` project, there are more than three HTML files due to the User Task generated forms.